### PR TITLE
[Improvement](memory) clear arena when finalize one row

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
clear arena when finalize one row

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/olap/memtable.cpp [hunk 1]
```diff
@@ -373,19 +373,29 @@ void MemTable::_finalize_one_row(RowInBlock* row,
...
         }
+
+        _arena->clear();
+
         if constexpr (is_final) {
             row->remove_init_agg();
+        } else {
+            for (size_t i = _tablet_schema->num_key_columns(); i < _num_columns; ++i) {
+                auto function = _agg_functions[i];
+                auto* agg_place = row->agg_places(i);
```
Comment (yiguolei) [inline be/src/olap/memtable.cpp:394]: why move code from line 383 to here??

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by w41ter — APPROVED** [review]

**Comment by yiguolei** [comment]
Very very important work, This PR will reduce many memory for agg table.

**Pull request merged**